src/views/ApartmentApp/index.js

```diff
@@ -1,96 +1,105 @@
-import React, { Component } from 'react';
+//Main apartment application page
+import React, { useState, useEffect } from 'react';
 import { Grid, Card, CardContent, Button } from '@material-ui/core/';
 import GordonLoader from '../../components/Loader';
 import StudentApplication from './components/StudentApplication';
 import StaffMenu from './components/StaffMenu';
 import user from '../../services/user';
-// import housing from '../../services/housing';
+import housing from '../../services/housing';
 import './apartmentApp.scss';
 
-export default class ApartApp extends Component {
-  constructor(props) {
-    super(props);
-    this.peopleSearch = React.createRef();
-    this.state = {
-      isStu: Boolean,
-      isHousingStaff: Boolean,
-      loading: true,
-      network: 'online',
-      userProfile: {},
-    };
-  }
+const ApartApp = (props) => {
+  const [loading, setLoading] = useState(true);
+  const [network, setNetwork] = useState('online');
+  const [userProfile, setUserProfile] = useState({});
+  const [canUseStaff, setCanUseStaff] = useState(false);
+  const [isUserStudent, setIsUserStudent] = useState(false);
 
-  componentDidMount() {
-    this.loadProfile();
-  }
+  useEffect(() => {
+    loadUserProfile();
+    checkHousingStaff();
+  });
```

Comment:
Not passing a dependency array means that this useEffect gets run on every render. I think you probably mean for these to get called only after the first render, which would require an empty dependency array as the second argument of useEffect.

Reply:
Thank you for clarifying that, I was unsure about how those dependencies worked and I meant to use it exactly the way you guessed it